### ISSUE TITLE
Support `#[nested]` for nested if-exprs, to only generate 1 enum type

### ISSUE THREAD

**Issue by Boscop** [taiki-e/auto_enums#64]
Thanks for this crate, it's very useful for me, I wish this impl  feature 
I haven't looked at your implementation but I think the point of `#[nested]` is to avoid generating another enum type for nested unification.
Currently it only seems to support match. Correct me if I'm wrong, but this will generate 2 enum types currently, right?
```rust
#[auto_enum(Iterator)]
let r = if start != self.pointer {
    #[auto_enum(Iterator)]
    let r = if foo {
        self.clip.events_window(self.pointer, dist).into_iter().map(|msg| msg.at(rel_to))
    } else {
        self.port_state
            .msgs_to2(&dst_state)
            .into_iter()
            .map(|msg| msg.at(rel_to))
    };
    self.pointer = start;
    r
} else {
    std::iter::empty()
};
```

If yes, please also add support for `#[nested]` to nested if-exprs, such that cases like this can use it, and only 1 enum type will be generated.

**Comment by taiki-e** [comment]
Thanks! I have filed #67 to support this.

**Comment by taiki-e** [comment]
Published 0.7.0 which fixes this issue. Thank you for your patience.
